plugins/speccy.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+
+from cloudbot import hook
+
+import re
+import requests
+
+from bs4 import BeautifulSoup as BS
+from contextlib import closing
+
+url_re = \
+    re.compile(
+        'https?:\/\/speccy\.piriform\.com\/results\/[a-zA-Z0-9]+', re.I)
+GPU_RE = \
+    re.compile(
+        '.*(amd|radeon|intel|integrated|nvidia|geforce|gtx).*\n.*', re.I)
+PICO_RE = \
+    re.compile(
+        '.*pico', re.I)
+KMS_RE = \
+    re.compile(
+        '.*kms', re.I)
+BOOSTER_RE = \
+    re.compile(
+        '.*booster', re.I)
+REVIVER_RE = \
+    re.compile(
+        '.*reviver', re.I)
+KILLER_RE = \
+    re.compile(
+        '.*killer.+service', re.I)
+
+
+@hook.regex(url_re)
+def get_speccy_url(match):
+
+    with closing(requests.get(match.group(), )) as response:
+        soup = BS(response.content, "lxml", from_encoding=response.encoding)
+
+    body = soup.body
+
+    data = []
+
+    os_spec = body.find("div", text='Operating System')
+    if os_spec is not None:
+        data.append(
+            "\x02OS:\x02" + " " + os_spec.next_sibling.next_sibling.text)
+
+    ram_spec = body.find("div", text='RAM')
+    if ram_spec is not None:
+        data.append(
+            "\x02RAM:\x02" + " " + ram_spec.next_sibling.next_sibling.text)
+
+    cpu_spec = body.find("div", text='CPU')
+    if cpu_spec is not None:
+        data.append(
+            "\x02CPU:\x02" + " " + cpu_spec.next_sibling.next_sibling.text)
```

Comment:
These three `if` block could most likely be generalized to a function call since they are almost exactly the same.

Reply:
I'll compress that too.